pss/trojan_chunk.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2020 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package pss
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"encoding/json"
+	"math/big"
+	"reflect"
+
+	"github.com/ethersphere/swarm/chunk"
+	"github.com/ethersphere/swarm/storage"
+)
+
+// TODO: can we re-use some existing types here?
+type trojanHeaders struct {
```

Comment:
you do not need these structs

Reply:
just have the message struct call it `message`
then you will have a `for` method on it which outputs a `chunk.Chunk`

Reply:
so, as a first implementation, i tried to model these structs in a way that they look similar to what's in the book, since this was the guide for coding this.

they can for sure be merged or combined into a fewer amount. but this was the easiest way for me to give it shape and understand which part of the trojan chunk goes where and does what.

the `toContentAddressedChunk` func takes a `trojanChunk` struct and outputs a `chunk.Chunk`. it serializes the `trojanMessage` and uses this as the chunk payload, and takes the trojan chunk address and uses it as the address for the regular chunk.

the reason why i chose to have an address as a field in the `trojanChunk` struct is to have it encapsulated so that the correct nonce can be found.

Reply:
based on our last coding session, i've reduced the implementation to use just a `trojanMessage` struct